test/e2e/examples_test.go

```diff
@@ -24,6 +24,42 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+func TestXonoticGameServerReady(t *testing.T) {
+	t.Parallel()
+	gs := &agonesv1.GameServer{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "xonotic-",
+		},
+
+		Spec: agonesv1.GameServerSpec{
+			Container: "xonotic",
+			Ports: []agonesv1.GameServerPort{{
+				ContainerPort: 26000,
+				Name:          "default",
+				PortPolicy:    agonesv1.Dynamic,
+				Protocol:      corev1.ProtocolUDP,
+			}},
+			Template: corev1.PodTemplateSpec{
```

Comment:
Let's add a delay to the health check like we do in the Fleet:

https://github.com/googleforgames/agones/blob/a92303b6f3a836e48352bd760eec1edec1be6756/examples/xonotic/fleet.yaml#L31-L33

Reply:
Looking at the test, the GameServer kicked over to Unhealthy, and I'm willing to bet this is why.

Reply:
Included the health check.

Reply:
Let's make `initialDelaySeconds` bigger 😃 I think image pulldown in the cluster that has a lot of traffic is taking longer than expected.

Let's try `300` seconds. 5 minutes.

Reply:
I'm willing to bet this will fail just like https://github.com/googleforgames/agones/pull/3498/files#r1390046433 , which is on Autopilot.

Let's try the same thing here, but with a CPU limit of  200m, and Memory of 700Mi - I had a look at the workloads in console, and I expect that will be more than enough 🤞🏻 we can always increase if need be.

Reply:
Included `resourses` in fleet.yaml, gameserver.yaml, and in this test.
Test fails - https://gist.github.com/Kalaiselvi84/d36cca998f12d29744a2cf65c6ad6ac3

Reply:
Every time I am facing this terminating state issue😕, I create a new cluster and install the agones-system🎮. As a result, all the tests pass✅. Are we good to merge this PR?

Reply:
Sure, Thanks!